scripts/srt-dev.lua

```diff
@@ -0,0 +1,887 @@
+local ext_type_select = {
+	[-1] = "SRT_CMD_NONE",
+	[0] = "none",
+	[1] = "SRT_CMD_HSREQ",
+	[2] = "SRT_CMD_HSRSP",
+	[3] = "SRT_CMD_KMREQ",
+	[4] = "SRT_CMD_KMRSP",
+	[5] = "SRT_CMD_SID",
+	[6] = "SRT_CMD_SMOOTHER"
+}
+fields.ext_type_msg_tree = ProtoField.none("srt_dev.ext_type", "Extension Type Message", base.NONE)
+fields.ext_type = ProtoField.uint16("srt_dev.ext_type", "Extension Type", base.HEX, ext_type_select, 0xF)
+fields.ext_size = ProtoField.uint16("srt_dev.ext_size", "Extension Size", base.DEC)
+
+-- Handshake packet, ext type == SRT_CMD_HSREQ or SRT_CMD_HSRSP filed
+fields.srt_version = ProtoField.uint32("srt_dev.srt_version", "SRT Version", base.HEX)
+fields.srt_flags = ProtoField.uint32("srt_dev.srt_flags", "SRT Flags", base.HEX)
+fields.tsbpb_resv = ProtoField.uint16("srt_dev.tsbpb_resv", "TsbPb Receive", base.DEC)
+fields.tsbpb_delay = ProtoField.uint16("srt_dev.tsbpb_delay", "TsbPb Delay", base.DEC)
+fields.rcv_tsbpb_delay = ProtoField.uint16("srt_dev.rcv_tsbpb_delay", "Receive TsbPd Delay", base.DEC)
+fields.snd_tsbpb_delay = ProtoField.uint16("srt_dev.snd_tsbpb_delay", "Send TsbPd Delay", base.DEC)
+
+-- V adn PT status flag
+local V_state_select = {
+	[1] = "Initial version"
+}
+fields.V_state = ProtoField.uint8("srt_dev.V_state", "V", base.HEX, V_state_select, 0x70)
+local PT_state_select = {
+	[0] = "Reserved",
+	[1] = "MSmsg",
+	[2] = "KMmsg",
+	[7] = "Reserved to discriminate MPEG-TS packet(0x47=sync byte)"
+}
+fields.PT_state = ProtoField.uint8("srt_dev.PT_state", "PT", base.HEX, state_table, 0xF)
+fields.sign = ProtoField.uint16("srt_dev.sign", "Signature", base.HEX)
+local resv_select = {
+	[0] = "Reserved for flag extension or other usage"
+}
+fields.resv = ProtoField.uint8("srt_dev.resv", "Resv", base.DEC, state_table, 0xFC)
+fields.ext_KK_state = ProtoField.uint8("srt_dev.ext_KK_state", "KK_state", base.HEX, KK_state_select, 0x3)
+fields.KEKI = ProtoField.uint32("srt_dev.KEKI", "KEKI", base.DEC)
+fields.cipher = ProtoField.uint8("srt_dev.cipher", "Cipher", base.DEC)
+fields.auth = ProtoField.uint8("srt_dev.auth", "auth", base.DEC)
+fields.SE = ProtoField.uint8("srt_dev.SE", "SE", base.DEC)
+fields.resv1 = ProtoField.uint8("srt_dev.resv1", "resv1", base.DEC)
+fields.resv2 = ProtoField.uint16("srt_dev.resv2", "resv2", base.DEC)
+fields.slen = ProtoField.uint8("srt_dev.slen", "Salt length(bytes)/4", base.DEC)
+fields.klen = ProtoField.uint8("srt_dev.klen", "SEK length(bytes)/4", base.DEC)
+fields.salt = ProtoField.uint32("srt_dev.salt", "Salt key", base.DEC)
+fields.wrap = ProtoField.none("srt_dev.wrap", "Wrap key(s)", base.NONE)
+
+-- Handshake packet, ext_type == SRT_CMD_SMOOTHER filed
+fields.smoother_block = ProtoField.none("srt_dev.smoother_block", "Smoother Block", base.NONE)
+
+-- Handshake packet, ext_type == SRT_CMD_SID filed
+fields.sid_block = ProtoField.none("srt_dev.sid_block", "SID Block", base.NONE)
+
+-- SRT flags
+local flag_state_select = {
```

Comment:
I personally prefer displaying flags in the form of, for example, `+TSBPDSND -NAKREPORT +REXMITFLG`.